### PR TITLE
better error handling in getConversationInfo

### PR DESCRIPTION
**Pull request by mmaxim**
@songgao r?

You get this display now:

```
[mike@lisa-keybase]-[~/go/src/github.com/keybase/client/go] (mike/errorcleanup)$ kb chat list
▶ WARNING Running in devel mode
▶ WARNING failed to unbox message: convID: 12385818184558444544 msgID: 2 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 12385818184558444544 msgID: 1 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 10916103195311276032 msgID: 3 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 10916103195311276032 msgID: 1 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 8237287940140040192 msgID: 2 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 8237287940140040192 msgID: 1 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 11058899962229489664 msgID: 2 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 11058899962229489664 msgID: 1 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 3984166482157240320 msgID: 4 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
▶ WARNING failed to unbox message: convID: 3984166482157240320 msgID: 1 err: error unboxing chat message: box.Open failure; ciphertext was corrupted or wrong key
[1]  ??? [???] no valid message in the conversation: convID: 10916103195311276032
[2]  ??? [???] no valid message in the conversation: convID: 3984166482157240320
[3]  ??? [???] no valid message in the conversation: convID: 12385818184558444544
[4]  ??? [???] no valid message in the conversation: convID: 11058899962229489664
[5]  ??? [???] no valid message in the conversation: convID: 8237287940140040192
```

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits 53a567080cb8...77dc6f72a351 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475272877629-53a567080cb8-77dc6f72a351-linux/New_Proof_PGP_generating-Generating_PGP-53a567080cb8.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475272877629-53a567080cb8-77dc6f72a351-linux/New_Proof_PGP_generating-Generating_PGP-77dc6f72a351.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
</summary>

File: go/service/chat_local.go [hunk 1]
```diff
@@ -537,7 +539,7 @@ func (h *chatLocalHandler) getConversationInfo(ctx context.Context,
 	conversationInfo.TopicType = conversationRemote.Metadata.IdTriple.TopicType
 
 	if len(conversationRemote.MaxMsgs) == 0 {
-		return chat1.ConversationInfoLocal{}, nil,
+		return conversationInfo, maxMessages,
```
Comment (songgao) [inline go/service/chat_local.go]: Is there a reason to return existing variables here? I just changed all of them to newly constructed zero values per patrick's suggesting in [a previous PR](https://github.com/keybase/client/pull/4405#pullrequestreview-1791423)

Comment (mmaxim) [inline go/service/chat_local.go]: I just didn't want to return `nil` for `maxMessages`, returning blank would have been fine too.

Comment (mmaxim) [inline go/service/chat_local.go]: I guess it doesn't matter that much, in general I don't like those named return values either, but I saw they were already in there.

Comment (mmaxim) [inline go/service/chat_local.go]: I'll change it back.

Comment (songgao) [inline go/service/chat_local.go]: Sounds good. Thanks Mike

**Comment by mmaxim** [comment]
Also, I got into this state on dev because I restarted md server, and all the crypt keys changed.

**Comment by songgao** [comment]
Maybe docker didn't persist MD server's storage across restarts?

**Comment by mmaxim** [comment]
Yeah it doesn't, I just exploited that case for improving the error behavior.

**Pull request merged**